autoload/converter/slc.vim

```
@@ -0,0 +1,59 @@
+" A zip file of many collections of sokoban levels can be downloaded from
+" here: http://www.sourcecode.se/sokoban/download/Levels. They are XML files
+" with the extension "slc". The converter#slc#ToJSON() function does a good
+" job of converting them to JSON format with mostly search and replace
+" commands. You will need to replace any XML character entity references:
+" &apos; &quot; &amp; &lt; &gt;. The rest of the file should be OK, but it
+" doesn't hurt to give it a once-over to be sure.
+
+" This function assumes the following workflow:
+"       :e FooBar.json
+"       :r Foobar.slc
+"       :call SLCtoJSON()
+function! converter#slc#ToJSON()
+    " Outer tag becomes start/end of JSON object.
+    %s/<SokobanLevels.*/{/
+    %s/<\/SokobanLevels>/}/
+
+    " Uppermost elements to JSON data.
+    %s/<Title>\(.*\)<\/Title>/"title":"\1",/
+    %s/<Email>\(.*\)<\/Email>/"email":"\1",/
+    %s/<Url>\(.*\)<\/Url>/"url":"\1",/
```

Suppress the errors if nothing found.
```suggestion
    %s/<Email>\(.*\)<\/Email>/"email":"\1",/e
    %s/<Url>\(.*\)<\/Url>/"url":"\1",/e
```